ui/components/component-library/avatar-token/avatar-token.js

```diff
@@ -0,0 +1,125 @@
+import React, { useState, useEffect } from 'react';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import Box from '../../ui/box/box';
+import { BaseAvatar } from '../base-avatar';
+
+import {
+  COLORS,
+  SIZES,
+  DISPLAY,
+  ALIGN_ITEMS,
+  JUSTIFY_CONTENT,
+} from '../../../helpers/constants/design-system';
+
+export const AvatarToken = ({
+  size = SIZES.MD,
+  tokenName,
+  tokenImageUrl,
+  showHalo,
+  color = COLORS.TEXT_DEFAULT,
+  backgroundColor = COLORS.BACKGROUND_ALTERNATIVE,
+  borderColor = COLORS.TRANSPARENT,
+  className,
+  ...props
+}) => {
+  const [showFallback, setShowFallback] = useState(false);
+
+  useEffect(() => {
+    setShowFallback(!tokenImageUrl);
+  }, [tokenImageUrl]);
+
+  const handleOnError = () => {
+    setShowFallback(true);
+  };
+
+  const fallbackString = tokenName && tokenName[0] ? tokenName[0] : '?';
+
+  return (
+    <BaseAvatar
+      size={size}
+      display={DISPLAY.FLEX}
+      alignItems={ALIGN_ITEMS.CENTER}
+      justifyContent={JUSTIFY_CONTENT.CENTER}
+      className={classnames(
+        'avatar-token',
+        showHalo && 'avatar-token--with-halo',
+        className,
+      )}
+      {...{ backgroundColor, borderColor, color, ...props }}
+    >
+      {showFallback ? (
+        fallbackString
+      ) : (
+        <>
+          {showHalo && (
+            <img
+              src={tokenImageUrl}
+              className={showHalo ? 'avatar-token__token-image--blurred' : ''}
+              aria-hidden="true"
+            />
+          )}
+          <img
+            className={
+              showHalo
+                ? 'avatar-token__token-image--size-reduced'
+                : 'avatar-token__token-image'
+            }
+            onError={handleOnError}
+            src={tokenImageUrl}
+            alt={tokenName || 'token avatar'}
+          />
+        </>
+      )}
+    </BaseAvatar>
+  );
+};
+
+AvatarToken.propTypes = {
+  /**
+   * The tokenName accepts the string to render the first letter of the AvatarToken. This will be used as the fallback display if no image url is passed to the tokenImageUrl
+   */
+  tokenName: PropTypes.string,
+  /**
+   * The tokenImageUrl accepts the string of the image to be rendered
+   */
+  tokenImageUrl: PropTypes.string,
+  /**
+   * The showHalo accepts a boolean prop to render the image with halo effect
+   */
+  showHalo: PropTypes.bool,
+  /**
+   * The size of the AvatarToken.
+   * Possible values could be 'SIZES.XS', 'SIZES.SM', 'SIZES.MD', 'SIZES.LG', 'SIZES.XL'
+   * Defaults to SIZES.MD
+   */
+  size: PropTypes.oneOf(Object.values(SIZES)),
+  /**
+   * The children to be rendered inside the AvatarToken
+   */
+  children: PropTypes.node,
```

Comment:
This is my bad this component shouldn't have any children. We might in future need a way of refining the prop types that we spread from `...Box.propTypes` but we can have a discussion about it and do that in another PR

Something like
```
const { margin, marginTop, marginRight, marginBottom, marginLeft, as, display } = Box.propTypes;

// In AvatarToken.propTypes
...{ margin, marginTop, marginRight, marginBottom, marginLeft, as, display }
```

Remove children from propTypes

```suggestion
```